firstvoices/backend/views/dictionary_views.py

```diff
@@ -1,30 +1,55 @@
-from drf_spectacular.utils import extend_schema, extend_schema_view
+from django.core.exceptions import PermissionDenied
+from django.http import Http404
+from drf_spectacular.utils import OpenApiResponse, extend_schema, extend_schema_view
 from rest_framework.viewsets import ModelViewSet
 
-from backend.serializers.dictionary_serializers import (
-    PartOfSpeech,
-    PartsOfSpeechSerializer,
-)
+from backend.models import DictionaryEntry, Site
+from backend.predicates import utils
+from backend.serializers.dictionary_serializers import DictionaryEntryDetailSerializer
 from backend.views.base_views import FVPermissionViewSetMixin
 
 
 @extend_schema_view(
     list=extend_schema(
-        description="A list of different parts-of-speech available that different words/phrases can be assigned to.",
-        responses={200: PartsOfSpeechSerializer},
+        description="A list of available dictionary entries for the specified site.",
+        responses={
+            200: DictionaryEntryDetailSerializer,
+            403: OpenApiResponse(description="Todo: Not authorized for this Site"),
+            404: OpenApiResponse(description="Todo: Site not found"),
+        },
     ),
     retrieve=extend_schema(
-        description="Details about a specific part of speech.",
-        responses={200: PartsOfSpeechSerializer},
+        description="A dictionary entry from the specified site.",
+        responses={
+            200: DictionaryEntryDetailSerializer,
+            403: OpenApiResponse(description="Todo: Error Not Authorized"),
+            404: OpenApiResponse(description="Todo: Not Found"),
+        },
     ),
 )
-class PartsOfSpeechViewSet(FVPermissionViewSetMixin, ModelViewSet):
+class DictionaryViewSet(FVPermissionViewSetMixin, ModelViewSet):
+    """
+    Dictionary entry information.
+    """
+
     http_method_names = ["get"]
-    serializer_class = PartsOfSpeechSerializer
-    queryset = PartOfSpeech.objects.prefetch_related("children").all()
-
-    @staticmethod
-    def get_list_queryset():
-        return PartOfSpeech.objects.prefetch_related("children").exclude(
-            parent__isnull=False
-        )
+    serializer_class = DictionaryEntryDetailSerializer
+
+    def get_validated_site(self):
```

Comment:
I believe this method will be reused. I'm using it in Categories API but it may also come handy in other nested routes api as well. Maybe we can move this to utilities or create a mixin for this which can be inherited if required ?